backend/mem/fileSystem.go

```diff
@@ -0,0 +1,78 @@
+package mem
+
+import (
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/backend"
+	"github.com/c2fo/vfs/v4/utils"
+	"path"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+var systemMap map[string]*File
+var fileList []*File
+
+// FileSystem implements vfs.Filesystem for the mem filesystem.
+type FileSystem struct{}
+
+// FileSystem will return a retrier provided via options, or a no-op if none is provided.
+func (fs *FileSystem) Retry() vfs.Retry {
+	return vfs.DefaultRetryer()
+}
+
+// NewFile function returns the mem implementation of vfs.File.  NOT DONE
+func (fs *FileSystem) NewFile(volume string, name string) (vfs.File, error) {
+
+	if !path.IsAbs(name) {
+		name = path.Join("/", name)
+	}
```

Comment:
We cant just assume they meant to have a leading slash.  In fact that might be really dangerous.
If a user somehow expected relative path, and they put 
```
file, _ := fs.Newfile("", "etc/passwd")
file.Delete()
```
it would delete /etc/passwd.

Instead we should be doing validation and return an error if necessary.

Reply:
I'm going to play devil's advocate here and disagree. Working at the file system level should assume you are using the volume as your starting point, because it has no location context.

Just slightly modifying your example for additional context demonstrates this:

```
file, _ := fs.Newfile("C:", "etc/passwd")
file.Delete()
```

It should be up to the developer to do an "Exist()" check on the returned file before deleting. Some operating systems won't blink an eye or ask twice if you type:

`rm -rf /`

I don't think our file system implementations should behave any different. This is something we should probably clarify in the interface documentation.